### ISSUE TITLE
Feature request: update a single query's hint with --admit_except and --record_hints

### ISSUE THREAD

**Issue by tchajed** [FStarLang/FStar#1139]
It would be nice if `--admit_except <id> --record_hints` updated an existing set of hints for a particular id without removing the rest. Currently the behavior is to overwrite all the hints with a single one, which isn't terribly useful.

**Comment by nikswamy** [comment]
Sounds useful. @wintersteiger ?

**Comment by wintersteiger** [comment]
Absolutely, we definitely want that to work! Did you try with both `--record_hints` and `--use_hints`? When both are on it should at least _load_ all the hints, but I think it will then save all of them (including the updated one) too.

**Comment by s-zanella** [comment]
@wintersteiger Unfortunately that won't work: using `--record_hints --use_hints --admit_except ...` will load all the hints, but it will write a new hints file containing hints only for queries that weren't admitted.
